test/components/views/rooms/SearchBar-test.tsx

```diff
@@ -40,95 +35,65 @@ jest.mock("../../../../src/KeyBindingsManager", () => ({
-        get Search() { // eslint-disable-line @typescript-eslint/naming-convention
-            return mockWarningKind;
-        },
-    },
-    default: jest.fn(({ children }) => (
-        <div>{ children }</div>
-    )),
-}));
-
-/** mock out PosthogTrackers component so it doesn't affect the result of our test */
-jest.mock('../../../../src/PosthogTrackers', () => ({
-    __esModule: true,
-    PosthogScreenTracker: jest.fn(({ children }) => (
-        <div>{ children }</div>
-    )),
-}));
-
 describe("SearchBar", () => {
-    beforeEach(() => {
-        wrapper = mount(<SearchBar {...searchProps} />);
-    });
-
     afterEach(() => {
-        wrapper.unmount();
         searchProps.onCancelClick.mockClear();
         searchProps.onSearch.mockClear();
     });
 
-    it("must render child components and pass necessary props", () => {
-        const postHogScreenTracker = wrapper.find(PosthogScreenTracker);
-        const searchWarning = wrapper.find(SearchWarning);
-
-        expect(postHogScreenTracker.length).toBe(1);
-        expect(searchWarning.length).toBe(1);
-        expect(postHogScreenTracker.props().screenName).toEqual("RoomSearch");
-        expect(searchWarning.props().isRoomEncrypted).toEqual(searchProps.isRoomEncrypted);
-        expect(searchWarning.props().kind).toEqual(mockWarningKind);
-    });
-
     it("must not search when input value is empty", () => {
-        const roomButtons = wrapper.find(".mx_SearchBar_button");
-        const searchButton = wrapper.find(".mx_SearchBar_searchButton");
+        const { container } = render(<SearchBar {...searchProps} />);
+        const roomButtons = container.querySelectorAll(".mx_SearchBar_button");
+        const searchButton = container.querySelectorAll(".mx_SearchBar_searchButton");
 
-        expect(roomButtons.length).toEqual(4);
+        expect(roomButtons.length).toEqual(2);
 
-        searchButton.at(0).simulate("click");
-        roomButtons.at(0).simulate("click");
-        roomButtons.at(2).simulate("click");
+        fireEvent.click(searchButton[0]);
+        fireEvent.click(roomButtons[0]);
+        fireEvent.click(roomButtons[1]);
 
         expect(searchProps.onSearch).not.toHaveBeenCalled();
     });
 
     it("must trigger onSearch when value is not empty", () => {
+        const { container } = render(<SearchBar {...searchProps} />);
         const searchValue = "abcd";
 
-        const roomButtons = wrapper.find(".mx_SearchBar_button");
-        const searchButton = wrapper.find(".mx_SearchBar_searchButton");
-        const input = wrapper.find(".mx_SearchBar_input input");
-        input.instance().value = searchValue;
+        const roomButtons = container.querySelectorAll(".mx_SearchBar_button");
+        const searchButton = container.querySelectorAll(".mx_SearchBar_searchButton");
+        const input = container.querySelector<HTMLInputElement>(".mx_SearchBar_input input");
+        if (input) {
+            input.value = searchValue;
+        }
```

Comment:
I think this if-check muddies the possible error. If the input is missing we'd say one of the later assertions fail, whereas strategically using `!` here would say cannot set value of undefined